Grow.c

```diff
@@ -3761,23 +3786,20 @@ int reshape_container(char *container, char *devname,
 	 */
 	ping_monitor(container);
 
-	if (!forked && !c->freeze_reshape)
-		if (continue_via_systemd(container, GROW_SERVICE, NULL))
-			return 0;
-
-	switch (forked ? 0 : fork()) {
+	/* Do not start a service if reshape frozen. */
+	switch (handle_forking(forked, container, !c->freeze_reshape)) {
```

Comment:
`!c->freeze_reshape` is totally not understandable now.

we cannot use systemd but we may still fork. Please unify those ways.